test/kehaar_component/core_test.clj

```diff
@@ -0,0 +1,139 @@
+(ns kehaar-component.core-test
+  (:require
+    [clojure.core.async :as async]
+    [clojure.test :refer [is deftest testing use-fixtures]]
+    [com.stuartsierra.component :as component]
+    [kehaar-component.events-exchange :as ee]
+    [kehaar-component.external-service :as es]
+    [kehaar-component.incoming-event-service :as ies]
+    [kehaar-component.incoming-service :as is]
+    [kehaar-component.kehaar-rabbitmq :as krmq]
+    [kehaar-component.outgoing-event-service :as oes]
+    [kehaar-component.system :as system]
+    [langohr.core :as rmq]))
+
+;;
+;; These tests require RabbitMQ. An easy way to get it up and running
+;; and available at the config set in the rmq-config below if you have
+;; docker installed is
+;;
+;;
+;; $ docker pull rabbitmq
+;; $ docker run -d -p 127.0.0.1:5672:5672 -p 127.0.0.1:15672:15672 \
+;;     --hostname my-rabbit --name some-rabbit rabbitmq:3-management
+;;
+
+(def rmq-config
+  {:connection {:host "localhost"
+                :port 5672}
+   :max-retries 5})
+
+(def incoming-service-config
+  {:config {:type :incoming
+            :queue-name "direct-exchange-queue-name"
+            :handler-fn #(-> % (assoc :status :ok))
+            :exclusive false
+            :durable true
+            :auto-delete false}})
+
+(def external-service-config
+  (-> incoming-service-config
+      (assoc-in [:config :type] :external)
+      (update :config dissoc :handler-fn)))
+
+(def incoming-events-stub-chan1 (async/chan))
+(def incoming-events-stub-chan2 (async/chan))
+
+(def incoming-event-service1-config
+  {:config {:type :incoming-event
+            :queue-name "topic-exchange-consume-queue1"
+            :handler-fn #(async/put! incoming-events-stub-chan1 %)
+            :routing-key "test.events"
+            :exclusive false
+            :durable true
+            :auto-delete false}})
+
+(def incoming-event-service2-config
+  {:config {:type :incoming-event
+            :queue-name "topic-exchange-consume-queue2"
+            :handler-fn #(async/put! incoming-events-stub-chan2 %)
+            :routing-key "test.events"
+            :exclusive false
+            :durable true
+            :auto-delete false}})
+
+(def outgoing-event-service-config
+  (-> incoming-event-service1-config
+      (assoc :queue-name "topic-exchange-produce-queue")
+      (assoc-in [:config :type] :outgoing-event)
+      (update :config dissoc :handler-fn)))
+
+
+(deftest ^:rabbitmq kehaar-component-services
```

Comment:
Let's add test selectors to the project.clj so that `lein test` doesn't run the rabbitmq tests, `lein test :rabbitmq` just runs the rabbitmq tests, and`lein test :all` runs everything.